monai/networks/trt_compiler.py

```diff
@@ -0,0 +1,545 @@
+        Args:
+             input_example: passed to onnx.export()
+        """
+
+        if self.engine is not None:
+            return
+
+        export_args = self.export_args
+        dbs = self.dynamic_batchsize
+        if dbs:
+            if len(self.profiles) > 0:
+                raise ValueError("ERROR: Both dynamic_batchsize and input_profiles set for TrtCompiler!")
+            if len(dbs) != 3:
+                raise ValueError("dynamic_batchsize has to have len ==3 ")
+            profiles = {}
+            for id, val in input_example.items():
+                sh = val.shape[1:]
+                profiles[id] = [[dbs[0], *sh], [dbs[1], *sh], [dbs[2], *sh]]
+            self.profiles = [profiles]
+
+        if len(self.profiles) > 0:
+            export_args.update({"dynamic_axes": get_dynamic_axes(self.profiles)})
+
+        add_casts_around_norms(model)
+
+        if self.method == "torch_trt":
+            enabled_precisions = [torch.float32]
+            if self.precision == "fp16":
+                enabled_precisions.append(torch.float16)
+            elif self.precision == "bf16":
+                enabled_precisions.append(torch.bfloat16)
+            inputs = list(input_example.values())
+            ir_model = convert_to_torchscript(model, inputs=inputs, use_trace=True)
+            engine_bytes = torch_tensorrt.convert_method_to_trt_engine(
+                ir_model,
+                "forward",
+                inputs=inputs,
+                ir="torchscript",
+                enabled_precisions=enabled_precisions,
+                **export_args,
+            )
+        else:
+            if self.method == "onnx_dynamo":
+                import torch_onnx
+
+                torch_onnx.patch_torch()
+                export_args["dynamo"] = True
+
+            # Use temporary directory for easy cleanup in case of external weights
+            with tempfile.TemporaryDirectory() as tmpdir:
+                onnx_path = Path(tmpdir) / "model.onnx"
+                self.logger.info(
+                    f"Exporting to {onnx_path}:\n\toutput_names={self.output_names}\n\texport args: {export_args}"
+                )
+                convert_to_onnx(
+                    model,
+                    input_example,
+                    filename=str(onnx_path),
+                    input_names=self.input_names,
+                    output_names=self.output_names,
+                    **export_args,
+                )
+                self.logger.info("Export to ONNX successful.")
+                engine_bytes = self._onnx_to_trt(str(onnx_path))
+
+        open(self.plan_path, "wb").write(engine_bytes)
+
+
+def trt_forward(self, *argv, **kwargs):
+    """
+    Patch function to replace original model's forward() with.
+    Redirects to TrtCompiler.forward()
+    """
+    return self._trt_compiler.forward(self, argv, kwargs)
+
+
+def trt_compile(
+    model: torch.nn.Module,
+    base_path: str,
```

Comment:
if base_path does not exist, this arg seems not used. Can we change the type ti base_path: str | None, and add some doc strings for this case?

Reply:
Well, it says so in the first line: "TRT plan(s) saved to "base_path[.submodule].plan" path.". I am extending the language anyway.